plugins/optimization/src/main/java/org/qbicc/plugin/opt/InitializedStaticFieldBasicBlockBuilder.java

```diff
@@ -0,0 +1,35 @@
+package org.qbicc.plugin.opt;
+
+import org.qbicc.context.CompilationContext;
+import org.qbicc.graph.BasicBlockBuilder;
+import org.qbicc.graph.DelegatingBasicBlockBuilder;
+import org.qbicc.graph.MemoryAtomicityMode;
+import org.qbicc.graph.StaticField;
+import org.qbicc.graph.Value;
+import org.qbicc.graph.ValueHandle;
+import org.qbicc.graph.literal.Literal;
+import org.qbicc.type.definition.element.FieldElement;
+
+public class InitializedStaticFieldBasicBlockBuilder extends DelegatingBasicBlockBuilder {
+    final CompilationContext ctxt;
+
+    public InitializedStaticFieldBasicBlockBuilder(final CompilationContext ctxt, final BasicBlockBuilder delegate) {
+        super(delegate);
+        this.ctxt = ctxt;
+    }
+
+    @Override
+    public Value load(ValueHandle handle, MemoryAtomicityMode mode) {
+        if (handle instanceof StaticField) {
+            final FieldElement fieldElement = ((StaticField) handle).getVariableElement();
+            if (fieldElement.isReallyFinal()) {
+                Value contents = fieldElement.getEnclosingType().load().getInitialValue(fieldElement);
+                if (contents instanceof Literal) {
```

Comment:
I believe this covers the object literal cases as well but would appreciate confirmation.  Is the check that `contents instanceof Literal` effectively an assertion?  What else would it be here for a static field?

Reply:
That makes sense - thanks Dave